File: DependencyInjection3/app/src/main/java/com/arcore/dependencyinjection/viewmodel/ViewModelFactory.java
package com.arcore.dependencyinjection.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class ViewModelFactory implements ViewModelProvider.Factory {
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {

        if(modelClass.isAssignableFrom(MyViewModel.class))
            return (T) new MyViewModel("Nilesh,Teji","Teji");
        else{
            return null;
        }
    }
}

File: DependencyInjection3/app/src/main/java/com/arcore/dependencyinjection/viewmodel/ViewModelFactory.kt
package com.arcore.dependencyinjection.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

public class ViewModelFactory(email :String,password:String):ViewModelProvider.Factory{

val email=email
val password=password

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
      if (modelClass.isAssignableFrom(MyViewModel::class.java)){
          return MyViewModel(email,password) as T
      }
        else{
         throw IllegalArgumentException("Incorrect Creditionals")
      }
    }

}